### PR TITLE
Fix Phase-1 Barrel Pixel Inner / Outer ladders numbering in several monitoring tools

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

In the (long) discussion begun at https://github.com/CMSTrackerDPG/Tasks/issues/1#issuecomment-1456164032 (and following), it became apparent that the convention used to define inner / outer Phase-1 Barrel Pixel ladders based on their offline numbering in several different places in `cmssw` has become wrong after the major geometry fix in PR https://github.com/cms-sw/cmssw/pull/18651, after which the right convention is:
   * evenly numbered ladder = inner ladder = flipped ladder (for BPix 1,2,3)
   * oddly numbered ladder = inner ladder = flipped ladder (for BPix 4)
   
Screenshot from 2023-03-08 11-22-51

to be compared with a ~recent (2018) pixel hadrography from the material budget analysis  
   
image

#### PR validation:

`cmssw` compiles. 

**We DO expect regressions**

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport PR, but it does need to be backported to at least 13_0_X for 2023 data-taking.

cc:
@sroychow @ferencek @mroguljic @tsusa @arossi83 @cardinia @antoniovagnerini

### CODE REVIEW

File: DQM/SiPixelPhase1Common/src/SiPixelCoordinates.cc [hunk 1]
```diff
@@ -156,7 +156,7 @@ int SiPixelCoordinates::half(const DetId& detid) {
 // Ladders have a staggered structure
 // Non-flipped ladders are on the outer radius
 // Phase 0: Outer ladders are odd for layer 1,3 and even for layer 2
-// Phase 1: Outer ladders are odd for layer 4 and even for layer 1,2,3
+// Phase 1: Outer ladders are odd for layer 1,2,4 and even for layer 4
```
Comment (ferencek) [inline DQM/SiPixelPhase1Common/src/SiPixelCoordinates.cc:159]:
```suggestion
// Phase 1: Outer ladders are odd for layer 1,2,3 and even for layer 4
```

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by tvami** [comment]
+1
 * tests pass 
 * changes are expected in the Jenkins tests

**Comment by mmusich** [comment]
So one thing that bugs me is that now after the "fix" the ROCs which are actually painted in the DQM "occupancy-style" maps in Bpix are swapped along the module "bisecting line" on the long module side 

image


the question is if the "drawing" of the ROCs with the marked ROC0 in certain locations was also "wrong" such that 2 mistakes made a right, or not....

**Comment by mmusich** [comment]
for example, here is a comparison from the payload inspector comparing the BPix maps that one gets from:
```bash
getPayloadData.py --plugin pluginSiPixelQuality_PayloadInspector --plot plot_SiPixelBPixQualityMap --tag SiPixelQuality_byPCL_prompt_v2 --input_params '{}' --time_type Lumi --iovs '{"start_iov": "1555143233372258", "end_iov": "1555143233372258"}' --db Prod --test ;    
``` 
PIcheck

dumping the `SiPixelQuality` payload I see e.g. for layer1:
```
detId:303063068 s_module: 3  s_ladder: -3 error type:3 BadRocs: OOOOOOOOOOOOXXXX
detId:303067148 s_module: -2  s_ladder: -4 error type:3 BadRocs: OOOOXXXXOOOOOOOO
detId:303071260 s_module: 3  s_ladder: -5 error type:3 BadRocs: OOOOOOXOOOOOOOOO
detId:303083536 s_module: -1  s_ladder: 5 error type:3 BadRocs: OOOOOOOOOOOOOOOX
detId:303087648 s_module: 4  s_ladder: 4 error type:3 BadRocs: XXXXOOOOOOOOOOOO
```

according to this, the single ROC masked in the module at (signed module = -1 , signed ladder = 5) should be on ROC0 (which is the one with the marked box). That was the case before the fix, but it's not anymore now.

**Comment by mmusich** [comment]
hold
   * https://github.com/cms-sw/cmssw/pull/40989#issuecomment-1460190875 has to be understood

**Comment by ferencek** [comment]
> for example, here is a comparison from the payload inspector comparing the
> BPix maps that one gets from:
>
> getPayloadData.py --plugin pluginSiPixelQuality_PayloadInspector --plot plot_SiPixelBPixQualityMap --tag SiPixelQuality_byPCL_prompt_v2 --input_params '{}' --time_type Lumi --iovs '{"start_iov": "1555143233372258", "end_iov": "1555143233372258"}' --db Prod --test ;
>
> [image: PIcheck]
> <https://user-images.githubusercontent.com/5082376/223730643-f4fa9a05-e5b9-4f57-a113-292b078f5bb3.gif>
>
> dumping the SiPixelQuality payload I see e.g. for layer1:
>
> detId:303063068 s_module: 3  s_ladder: -3 error type:3 BadRocs: OOOOOOOOOOOOXXXX
> detId:303067148 s_module: -2  s_ladder: -4 error type:3 BadRocs: OOOOXXXXOOOOOOOO
> detId:303071260 s_module: 3  s_ladder: -5 error type:3 BadRocs: OOOOOOXOOOOOOOOO
> detId:303083536 s_module: -1  s_ladder: 5 error type:3 BadRocs: OOOOOOOOOOOOOOOX
> detId:303087648 s_module: 4  s_ladder: 4 error type:3 BadRocs: XXXXOOOOOOOOOOOO
>
> according to this, the single ROC masked in the module at (signed module =
> -1 , signed ladder = 5) should be on ROC0 (which is the one with the marked
> box). That was the case before the fix, but it's not anymore now.
>

My naive guess would be that drawing of ROC0 somehow does rely on the code
modified in this PR while filling of the red ROCs does.

>

**Comment by mmusich** [comment]
> My naive guess would be that drawing of ROC0 somehow does rely on the code
modified in this PR while filling of the red ROCs does.

This does not make sense. I guess you missed a "not" somewhere. Anyhow the point is that the render plugin of the DQM GUI has to be updated in synch (that will utterly break the plots produced in the old versions though)

**Comment by ferencek** [comment]
> > My naive guess would be that drawing of ROC0 somehow does rely on the code
> > modified in this PR while filling of the red ROCs does.
> 
> This does not make sense. I guess you missed a "not" somewhere. Anyhow the point is that the render plugin of the DQM GUI has to be updated in synch (that will utterly break the plots produced in the old versions though)

Indeed, I was typing on my phone while commuting home and missed one "not". This was done by sending an email so I could not immediately edit as I am not logged in to GitHub on my phone. Sorry for the confusing message. I wonder if the DQM GUI can somehow have a version-dependent behavior.

**Comment by emanueleusai** [comment]
We do not have this functionality in the DQM GUI at the moment.

**Comment by emanueleusai** [comment]
Ah, that's clever. I believe that should work.

**Comment by mmusich** [comment]
with the last push I have removed the changes involving the DQM packages, since they have deeper consequences and need to be re-validate carefully. Those will be followed up in another PR.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by tvami** [comment]
maybe we should ask an explicit signature from tracker? I feel like this is not very `alca` specific as the PR would suggest right now

**Comment by tvami** [comment]
assign trk-dpg

**Comment by tvami** [comment]
+alca
 * tests pass

**Comment by mmusich** [comment]
@tvami, `db` signature is needed as well here. 
Thanks.

**Comment by tvami** [comment]
+db

**Comment by rappoccio** [comment]
+1

* Errors are due to read errors, unrelated to this PR.

**Comment by rappoccio** [comment]
please merge

**Pull request merged**